nsq/writer.py

```diff
@@ -143,9 +146,9 @@ def _pub(self, command, topic, msg, delay_ms=None, callback=None):
 
         try:
             conn.send(cmd(*args))
-        except Exception:
-            logger.exception('[%s] failed to send %s' % (conn.id, command))
+        except:
             conn.close()
+            raise PubFailure('[%s] failed to send %s' % (conn.id, command))
```

Comment:
same here, don't want to introduce new failure semantics, let's drop the `raise` please